src/mpid/ch4/netmod/ofi/util.c

```diff
@@ -108,7 +109,7 @@ uint64_t MPIDI_OFI_mr_key_alloc(int key_type, uint64_t requested_key)
         case MPIDI_OFI_COLL_MR_KEY:
             {
                 MPIR_Assert(requested_key != MPIDI_OFI_INVALID_MR_KEY);
-                ret_key = requested_key | (1ULL << MPIDI_OFI_MR_KEY_PREFIX_SHIFT);
```

Comment:
How about defining `MPIDI_OFI_MR_KEY_COLL_MASK` as one of the provider capabilities?

Reply:
Hmm, that'll be a macro in the capability set, right? What's the advantage of that?  

The one disadvantage I see with that approach is when the mr key size changes within CXI, we would have to update that. There is a plan to change the mr key size from 2 bytes to 4 bytes in the near future. Whereas, this approach takes care of it dynamically.

Reply:
Okay, I think I understand the code now. I am marking this one as resolved. I'll add another comment.